app/forms/providers/means/regular_income_form.rb

```diff
@@ -105,33 +105,35 @@ def destroy_legal_aid_application_transaction_types!
         legal_aid_application
           .legal_aid_application_transaction_types
           .credits
-          .where.not(transaction_type_id: transaction_type_ids)
+          .where.not(transaction_type_id: transaction_type_ids_to_keep)
           .destroy_all
       end
 
       def destroy_regular_income_transactions!
         legal_aid_application
           .regular_incomes
-          .without(existing_income_regular_transactions)
+          .where.not(transaction_type_id: transaction_type_ids_to_keep)
           .destroy_all
       end
 
       def destroy_cash_income_transactions!
         legal_aid_application
           .cash_transactions
           .credits
-          .where.not(transaction_type_id: transaction_type_ids)
+          .where.not(transaction_type_id: transaction_type_ids_to_keep)
           .destroy_all
       end
 
-      def existing_income_regular_transactions
-        legal_aid_application
-          .regular_incomes
-          .where(transaction_type_id: transaction_type_ids)
+      def transaction_type_ids_to_keep
+        transaction_type_ids.append(housing_benefit_id)
+      end
+
+      def housing_benefit_id
+        @housing_benefit_id || TransactionType.find_by(name: "housing_benefit")&.id
```

Comment:
What's the intention behind this comparison?

Reply:
was intended. to be memoization but could remove safe navigator and set test up with housing benefits

Reply:
essential

Reply:
done